packages/dds/tree/src/simple-tree/simpleSchemaToJsonSchema.ts

```diff
@@ -0,0 +1,167 @@
+	for (const allowedType of schema.allowedTypes) {
+		anyOf.push(createSchemaRef(allowedType));
+	}
+
+	return {
+		$defs: definitions,
+		anyOf,
+	};
+}
+
+function convertDefinitions(
+	definitions: ReadonlyMap<string, SimpleNodeSchema>,
+): Record<string, NodeJsonSchema> {
+	const result: Record<string, NodeJsonSchema> = {};
+	for (const [key, value] of definitions) {
+		result[key] = convertNodeSchema(value);
+	}
+	return result;
+}
+
+/**
+ * Private symbol under which the results of {@link convertNodeSchema} are cached on an input {@link SimpleNodeSchema}.
+ */
+const nodeJsonSchemaCacheSymbol = Symbol("nodeJsonSchemaCache");
+
+/**
+ * Converts an input {@link SimpleNodeSchema} to a {@link NodeJsonSchema}.
+ *
+ * @remarks Caches the result on the input schema for future calls.
+ */
+function convertNodeSchema(schema: SimpleNodeSchema): NodeJsonSchema {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	if ((schema as any)[nodeJsonSchemaCacheSymbol] !== undefined) {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		return (schema as any)[nodeJsonSchemaCacheSymbol] as NodeJsonSchema;
+	}
+
+	let result: NodeJsonSchema;
+	switch (schema.kind) {
+		case "array":
+			result = convertArrayNodeSchema(schema);
+			break;
+		case "leaf":
+			result = convertLeafNodeSchema(schema);
+			break;
+		case "map":
+			result = convertMapNodeSchema(schema);
+			break;
+		case "object":
+			result = convertObjectNodeSchema(schema);
+			break;
+		default:
+			throw new TypeError(`Unknown node schema kind: ${(schema as SimpleNodeSchema).kind}`);
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	(schema as any)[nodeJsonSchemaCacheSymbol] = result;
+	return result;
+}
+
+function convertArrayNodeSchema(schema: SimpleArrayNodeSchema): ArrayNodeJsonSchema {
+	const allowedTypes: JsonSchemaRef[] = [];
+	schema.allowedTypes.forEach((type) => {
+		allowedTypes.push(createSchemaRef(type));
+	});
+	return {
+		type: "array",
+		_kind: "array",
+		items: {
+			anyOf: allowedTypes,
+		},
+	};
+}
+
+function convertLeafNodeSchema(schema: SimpleLeafNodeSchema): LeafNodeJsonSchema {
+	if (schema.type === "fluid-handle") {
+		throw new Error("Fluid handles are not yet round-trip supported via JSON schema.");
+	}
+
+	return {
+		type: schema.type,
+		_kind: "leaf",
+	};
+}
+
+function convertObjectNodeSchema(schema: SimpleObjectNodeSchema): ObjectNodeJsonSchema {
+	const properties: Record<string, FieldJsonSchema> = {};
+	const required: string[] = [];
+	for (const [key, value] of Object.entries(schema.fields)) {
+		const anyOf: JsonSchemaRef[] = [];
+		for (const allowedType of value.allowedTypes) {
+			anyOf.push(createSchemaRef(allowedType));
+		}
+
+		properties[key] = {
+			anyOf,
+		};
+		if (value.kind === "required") {
+			required.push(key);
+		}
+	}
+	return {
+		type: "object",
+		_kind: "object",
+		properties,
+		required,
+		additionalProperties: false, // TODO: get allowance from schema policy
```

Comment:
@CraigMacomber Is there currently a way for me to access this policy information without having access to the schema factory?